src/lib/demux.rs

```diff
@@ -421,6 +421,20 @@ where
             let mut output_reads = vec![];
             let mut qual_counter = BaseQualCounter::new();
 
```

Comment:
My only concern is that this may be expensive, since this will be done for _every_ read.

Reply:
I totally agree and would be hesitant to merge this as is without some benchmarking.  A couple of alternative suggestions for @jiangweiyao and @omicsorama to weigh in on:

1. Would it be sufficient to check the either just the first record, or the first and last in each RecordSet, either here or perhaps upon construction of the `PerFastqRecordSet` (where we already validate that the record sets are the same length).
2. If (1) is not sufficient, we could also e.g. check every nth record here.  E.g. if we checked every 50th or 100th record that would catch the vast majority of issues with a lot less performance overhead.

More generally - what's the problem we are trying to solve?  Catching when a user mixes up FASTQ files can be done by checking first/last etc.  But if we're trying to catch broken upstream software that might occasionally emit reads in differing order that would dictate checking every read on the way through.

Reply:
See #27 where I added in checking just the first read in each file (implicitly does not allow empty input FASTQs).